controllers/storageclient_controller.go

```diff
@@ -90,6 +90,13 @@ func (s *StorageClientReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		}
 		return res
 	})
+
+	if err := mgr.GetCache().IndexField(context.TODO(), &v1alpha1.StorageClassClaim{}, storageClientAnnotationKey, func(obj client.Object) []string {
```

Comment:
I don't think we want a new context for each operation within `SetupWithManager` as well as we don't want to use  `context.TODO()` where `context.Background()` makes more sense. Let's initialize one context to be used within `SetupWithManager` and reuse it

Reply:
done.